src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionInfo.cs

```diff
@@ -39,17 +41,20 @@ public ConnectionInfo(ISqlConnectionFactory factory, string ownerUri, Connection
         /// <summary>
         /// Factory used for creating the SQL connection associated with the connection info.
         /// </summary>
-        public ISqlConnectionFactory Factory {get; private set;}
+        public ISqlConnectionFactory Factory { get; private set; }
 
         /// <summary>
         /// Properties used for creating/opening the SQL connection.
         /// </summary>
         public ConnectionDetails ConnectionDetails { get; private set; }
-        
+
         /// <summary>
-        /// The connection to the SQL database that commands will be run against.
+        /// A map containing all connections to the database that are associated with 
+        /// this ConnectionInfo's OwnerUri.
+        /// This is internal for testing access only
         /// </summary>
-        public DbConnection SqlConnection { get; set; }
+        internal readonly Dictionary<string, DbConnection> ConnectionTypeToConnectionMap =
```

Comment:
Any particular reason we went with a string for the key instead of having an enum?

Additionally, any chance we might need to make this a ConcurrentDictionary?